src/speakers/extraction.py

```diff
@@ -329,3 +332,339 @@ def get_speakers_in_bold(doc):
+    names=[]
+    for name in names_temp:
+        #if more than 40% of lines that could be names end with ':', filter lines that do not end with ':'
+        if count/len(names_temp)>0.40:
+            if name[-1]==':':
+                names.append(name)
+        #else filter lines that do end with ':', also filtering lines ending with '.' as they don't occur after a name
+        elif name[-1]!=':' and name[-1]!='.':
+            names.append(name)
+    return names
+
+
+def remove_last_char_if_colon(word):
+    if word[-1]==':' and len(word)!=0:
+        word=word[0:len(word)-1]
+    return word
+
+#below fucntion is to check how many words in the names to be filtered belong to a english dictionary and eliminate accordingly.
+def check_english_dict(names):
+    names_temp=names
+    names=[]
+    possible_names_in_english_dictionary=['Shah','Moderator','moderator','Raj','Participant']
+    for name in names_temp:
+        count=0
+        for word in name.split():
+            word=remove_last_char_if_colon(word)
+            if english_dictionary.check(word) and word not in possible_names_in_english_dictionary:
+                count=count+1
+            #condition added to deal with false positives
+            if name=='KFin':
+                count=count+1
+        #below condtion only accepts names that have less than 50% of the words belonging to a dictionary
+        if count/len(name.split())<=0.50:
+            names.append(name)
+    return names
+
+
+def check_repetations(names):
+    names_temp=names
+    for name_outer in names:
+        for name_inner in names:
+            if name_outer.find(name_inner)!=-1 and len(name_outer)>len(name_inner):
+                names_temp.remove(name_inner)
+    names=names_temp
+    return(names)
+
+
+def get_speaker_names(doc):
+    lines = get_lines(doc)
+    char_not_in_names=[',','-','&','?']
+    names=[]
+    order_of_speakers=[]
+    for line in lines:
+        is_name=0
+        words=line.split()
+        if len(words)<=5: 
+            #below condition eliminates lines that have only 1 word and contain less than four letters
+            if len(words)==1 and len(words[0])<4:
+                is_name=1
+            for word in words:
+                if not word[0].isupper():
+                    is_name=1
+                for letter in word:
+                    if letter.isnumeric() or letter in char_not_in_names:
+                        is_name=1
+            if is_name==0:
+                if line not in names:
+                    names.append(line)
+    names=check_english_dict(names)
+    names=check_last_char_if_colon(names)
+    names=check_repetations(names)
+    for line in lines:
+        if line in names:
+            order_of_speakers.append(line)
+    #order_of_speakers here has the repeated names of people in the order they speak.
+    return [names,lines,order_of_speakers]
+
+
+def get_conversation(doc):
+    names,lines,order_of_speakers=get_speaker_names(doc)
```

Comment:
If we aren't using `order_of_speakers` then we can use `_`.

Reply:
Okay